lib/OpenQA/WebAPI/Controller/Main.pm

```diff
@@ -24,7 +24,9 @@ use OpenQA::Schema::Result::Jobs;
 sub _group_result {
     my ($self, $group, $limit) = @_;
 
-    my $timecond = {">" => time2str('%Y-%m-%d %H:%M:%S', time - 24 * 3600 * 14, 'UTC')};
+    my $time_limit_days = $self->param('time_limit_days') // 14;
```

Comment:
Looking at it, param validation is something we are long overdue. Would Mojo recovers when you supply i.e. text instead of `\d+`?